File: mobile/ios/PrestamosApp/Models/Compra.swift
import Foundation

struct Compra: Codable, Identifiable {
    let id: Int
    let numeroFactura: String?
    let proveedorId: Int?
    let proveedorNombre: String?
    let fecha: String?
    let montoTotal: Double?
    let metodoPago: String?
    let estado: String?
    let descuento: Double?
    let impuesto: Double?
    
    enum CodingKeys: String, CodingKey {
        case id
        case numeroFactura = "numero_factura"
        case proveedorId = "proveedor_id"
        case proveedorNombre = "proveedor_nombre"
        case fecha
        case montoTotal = "monto_total"
        case metodoPago = "metodo_pago"
        case estado
        case descuento
        case impuesto
    }
}

struct ComprasResponse: Codable {
    let success: Bool
    let data: ComprasData?
}

struct ComprasData: Codable {
    let items: [Compra]
    let total: Int?
}

struct CompraResponse: Codable {
    let success: Bool
    let data: Compra
}

File: mobile/android/app/src/main/java/com/erp/prestamos/models/Compra.java
package com.erp.prestamos.models;

import com.google.gson.annotations.SerializedName;

public class Compra {
    @SerializedName("id")
    private int id;
    
    @SerializedName("numero_factura")
    private String numeroFactura;
    
    @SerializedName("proveedor_id")
    private int proveedorId;
    
    @SerializedName("proveedor_nombre")
    private String proveedorNombre;
    
    @SerializedName("fecha")
    private String fecha;
    
    @SerializedName("monto_total")
    private double montoTotal;
    
    @SerializedName("metodo_pago")
    private String metodoPago;
    
    @SerializedName("estado")
    private String estado;
    
    @SerializedName("descuento")
    private double descuento;
    
    @SerializedName("impuesto")
    private double impuesto;
    
    // Getters and Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    
    public String getNumeroFactura() { return numeroFactura; }
    public void setNumeroFactura(String numeroFactura) { this.numeroFactura = numeroFactura; }
    
    public int getProveedorId() { return proveedorId; }
    public void setProveedorId(int proveedorId) { this.proveedorId = proveedorId; }
    
    public String getProveedorNombre() { return proveedorNombre; }
    public void setProveedorNombre(String proveedorNombre) { this.proveedorNombre = proveedorNombre; }
    
    public String getFecha() { return fecha; }
    public void setFecha(String fecha) { this.fecha = fecha; }
    
    public double getMontoTotal() { return montoTotal; }
    public void setMontoTotal(double montoTotal) { this.montoTotal = montoTotal; }
    
    public String getMetodoPago() { return metodoPago; }
    public void setMetodoPago(String metodoPago) { this.metodoPago = metodoPago; }
    
    public String getEstado() { return estado; }
    public void setEstado(String estado) { this.estado = estado; }
    
    public double getDescuento() { return descuento; }
    public void setDescuento(double descuento) { this.descuento = descuento; }
    
    public double getImpuesto() { return impuesto; }
    public void setImpuesto(double impuesto) { this.impuesto = impuesto; }
}

